cincinnati/src/plugins/mod.rs

```diff
@@ -0,0 +1,285 @@
+//! The plugin defines and implements the plugin interface, conversion boilerplate,
+//! internal plugins, and web plugin helpers
+
+pub mod external;
+pub mod interface;
+pub mod internal;
+
+use crate as cincinnati;
+use failure::{Error, Fallible, ResultExt};
+use std::collections::HashMap;
+use try_from::{TryFrom, TryInto};
+
+/// Enum for the two IO variants used by InternalPlugin and ExternalPlugin respectively
+pub enum PluginIO {
+    InternalIO(InternalIO),
+    ExternalIO(ExternalIO),
+}
+
+/// Struct used by the ExternalPlugin trait impl's
+#[derive(Debug, Clone, PartialEq)]
+pub struct InternalIO {
+    pub graph: cincinnati::Graph,
+    pub parameters: HashMap<String, String>,
+}
+
+/// Struct used by the InternalPlugin trait impl's
+#[derive(Debug, Clone, PartialEq)]
+pub struct ExternalIO {
+    pub bytes: Vec<u8>,
+}
+
+/// Trait which fronts InternalPlugin and ExternalPlugin, allowing their trait objects to live in the same collection
+pub trait Plugin<T>
+where
+    // <T as TryInto<PluginIO>>::Err: std::error::Error + std::marker::Send + std::marker::Sync,
+    T: TryInto<PluginIO>,
+    T: TryFrom<PluginIO>,
+{
+    fn run(&self, t: T) -> Fallible<T>;
+}
+
+/// Trait to be implemented by internal plugins with their native IO type
+pub trait InternalPlugin {
+    fn run_internal(&self, InternalIO) -> Fallible<InternalIO>;
+}
+
+/// Trait to be implemented by external plugins with its native IO type
+///
+/// There's a gotcha in that this type can't be used to access the information
+/// directly as it's merely bytes.
+pub trait ExternalPlugin {
+    fn run_external(&self, ExternalIO) -> Fallible<ExternalIO>;
+}
+
+/// Dummy converter to satisfy the Trait
+impl TryFrom<PluginIO> for PluginIO {
+    type Err = Error;
+
+    fn try_from(io: Self) -> Fallible<Self> {
+        Ok(io)
+    }
+}
+
+/// Dummy convertert to satisfy the Trait
```

Comment:
Typo: `converter`. Also, you may want to align the docstrings for those three identity-converters.